openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -110,6 +111,23 @@
 # .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1718
 LOG_REQUEST_USER_CHANGES = getattr(settings, 'LOG_REQUEST_USER_CHANGES', False)
 
+# .. toggle_name: LOG_REQUEST_USER_CHANGE_HEADERS
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: False
+# .. toggle_description: Turn this toggle on to log all request headers, for all requests, for all user ids involved in
+#      any user id change detected by safe sessions. The headers will provide additional debugging information. The
+#      headers will be logged for all requests up until LOG_REQUEST_USER_CHANGE_HEADERS_DURATION seconds after
+#      the time of the last mismatch. The header details will be encrypted, and only available with the private key.
+# .. toggle_warnings: To work correctly, LOG_REQUEST_USER_CHANGES must be enabled and ENFORCE_SAFE_SESSIONS must be
```

Comment:
[suggestion, optional] May want to add a note that the logging volume could be high under some circumstances.

Reply:
I'm going to leave this off for now. We'll see how noisy it actually is.